.github/workflows/deploy-docker.yaml

```diff
@@ -0,0 +1,144 @@
+name: Build and Publish Docker image
```

Comment:
This should be split into two workflows. One to build and push the image, and another one to update the deployment spec.

Reply:
Will do this later